src/Microsoft.Azure.WebJobs.Host/Loggers/Logger/ApplicationInsights/TelemetryClientFactory.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.ApplicationInsights.Extensibility.Implementation;
+using Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector;
+using Microsoft.ApplicationInsights.Extensibility.PerfCounterCollector.QuickPulse;
+using Microsoft.ApplicationInsights.WindowsServer;
+using Microsoft.ApplicationInsights.WindowsServer.Channel.Implementation;
+using Microsoft.ApplicationInsights.WindowsServer.TelemetryChannel;
+
+namespace Microsoft.Azure.WebJobs.Host.Loggers
+{
+    /// <summary>
+    /// Creates a <see cref="TelemetryClient"/> for use by the <see cref="ApplicationInsightsLogger"/>. 
+    /// </summary>
+    [CLSCompliant(false)]
+    public class TelemetryClientFactory : ITelemetryClientFactory
+    {
+        /// <summary>
+        /// Creates a <see cref="TelemetryClient"/>. 
+        /// </summary>
+        /// <returns>The <see cref="TelemetryClient"/> instance.</returns>
+        public TelemetryClient Create(string instrumentationKey, SamplingPercentageEstimatorSettings samplingSettings)
+        {
+            TelemetryConfiguration config = InitializeConfiguration(instrumentationKey, samplingSettings);
+
+            TelemetryClient client = new TelemetryClient(config);
+
+            string assemblyVersion = GetAssemblyFileVersion(typeof(JobHost).Assembly);
+            client.Context.GetInternalContext().SdkVersion = $"webjobs: {assemblyVersion}";
+
+            return client;
+        }
+
+        internal static string GetAssemblyFileVersion(Assembly assembly)
+        {
+            AssemblyFileVersionAttribute fileVersionAttr = assembly.GetCustomAttribute<AssemblyFileVersionAttribute>();
+            return fileVersionAttr?.Version ?? LoggingConstants.Unknown;
+        }
+
+        internal static TelemetryConfiguration InitializeConfiguration(string instrumentationKey,
```

Comment:
I've added config settings to let you turn *some* stuff on/off (like sampling), with the idea that we'll open it up with more knobs in the future. We run the risk of having a very complicated config section, though.

As for adaptive sampling, we've been going back-and-forth. With it off, there's a pretty legitimate risk of someone turning on App Insights and hitting their limit within an hour. With it on, they'll lose some specific records and see partial data (Chris hit this in a demo, actually). So far we keep coming down on the side of "enabled by default" to prevent the throttling. But let me grab a few people and see if we can all agree.